feat(dotnet): document Android Native tombstone options (#18560)
getsentry/sentry-docs: docs/platforms/dotnet/guides/android/configuration/native-options.mdx

</SdkOption>

## Tombstones

> The [Tombstone Integration](/platforms/android/configuration/tombstones/) generates events for native crashes based on the tombstone data that Android attaches to `ApplicationExitInfo` on Android 12 and later. It requires Sentry Android SDK 8.32.0 or above.

Author: ```suggestion
The [Tombstone Integration](/platforms/android/configuration/tombstones/) generates events for native crashes based on the tombstone data that Android attaches to `ApplicationExitInfo` on Android 12 and later. It is available when using the Sentry SDK  for .NET version 6.7.0 or above.
```

> The [Tombstone Integration](/platforms/android/configuration/tombstones/) generates events for native crashes based on the tombstone data that Android attaches to `ApplicationExitInfo` on Android 12 and later. It is available when using the Sentry SDK  for .NET version 6.7.0 or above.

Reviewer: style: spacing ... absolute nitpicking

```suggestion
The [Tombstone Integration](/platforms/android/configuration/tombstones/) generates events for native crashes based on the tombstone data that Android attaches to `ApplicationExitInfo` on Android 12 and later. It is available when using the Sentry SDK for .NET version 6.7.0 or above.
```

<SdkOption name="Native.TombstoneEnabled" type="bool" defaultValue="false">

Enables native crash reporting via [tombstones](/platforms/android/configuration/tombstones/). Disabled by default.

</SdkOption>

<SdkOption name="Native.ReportHistoricalTombstones" type="bool" defaultValue="false">

> When enabled, the SDK reports all historical tombstones from Android's exit reasons list, not just the latest one. Disabled by default. Historical tombstones receive only minimal enrichment, as the available data is likely out of sync.

Reviewer: question: mention `6.7.0`

Should we also mention "Sentry SDK for .NET version 6.7.0 or above" to be consistent with `Native.TombstoneEnabled` ... or is it already implied with it's relation to `TombstoneEnabled`.

Author: I think it's implied - both of these options are part of the Tombstone integration and we explicitly state that integration is available when using version 6.7.0 or above.

Reviewer: ... and `ReportHistoricalTombstones` requires `TombstoneEnabled`.

> When enabled, the SDK reports all historical tombstones from Android's exit reasons list, not just the latest one. Disabled by default. Historical tombstones receive only minimal enrichment, as the available data is likely out of sync.

Reviewer: question: relation to `TombstoneEnabled`

Does `Native.ReportHistoricalTombstones` require `Native.TombstoneEnabled` to be enabled, or does it work independently of `Native.TombstoneEnabled`.

If dependent, it might be worth documenting that relationship ... otherwise it's OK, as I'd assume independence if not stated otherwise.

Author: Good call - `ReportHistoricalTombstones` has no effect unless `TombstoneEnabled` is also enabled.

Tweaked the docs to make that clear in 9d79a83c2.

> See [Historical Tombstones](/platforms/android/configuration/tombstones/#historical-tombstones).

Reviewer: suggestion: consistent "default" description

To be consistent with `Native.TombstoneEnabled`, I suggest moving "Disabled by default" to the equivalent location after the option, rather than before:

```suggestion
When enabled, the SDK reports all historical tombstones from Android's exit reasons list, not just the latest one. Historical tombstones receive only minimal enrichment, as the available data is likely out of sync.

See [Historical Tombstones](/platforms/android/configuration/tombstones/#historical-tombstones). Disabled by default.
```
